pkg/kafka/configkafka/config.go

```diff
@@ -363,18 +365,23 @@ type SASLConfig struct {
 	Username string `mapstructure:"username"`
 	// Password to be used on authentication
 	Password string `mapstructure:"password"`
-	// SASL Mechanism to be used, possible values are: (PLAIN, AWS_MSK_IAM_OAUTHBEARER, SCRAM-SHA-256 or SCRAM-SHA-512).
+	// SASL Mechanism to be used, possible values are: (PLAIN, AWS_MSK_IAM_OAUTHBEARER, OIDCFILE,
+	// OIDC_STRING, SCRAM-SHA-256 or SCRAM-SHA-512).
 	Mechanism string `mapstructure:"mechanism"`
 	// SASL Protocol Version to be used, possible values are: (0, 1). Defaults to 0.
 	Version int `mapstructure:"version"`
 	// AWSMSK holds configuration specific to AWS MSK.
 	AWSMSK AWSMSKConfig `mapstructure:"aws_msk"`
+	// OIDCFile holds configuration for OIDC with file-based secret
+	OIDCFILE OIDCFileConfig `mapstructure:"oidc_file"`
 }
 
 func (c SASLConfig) Validate() error {
 	switch c.Mechanism {
 	case "AWS_MSK_IAM_OAUTHBEARER":
 		// TODO validate c.AWSMSK
+	case "OIDCFILE":
+		// TODO valid c.OIDCFile
```

Comment:
This should be done as part of the change, not after since it can be partially released to users.